pandas/_libs/lib.pyx

```diff
@@ -1944,7 +1928,7 @@ cdef class StringValidator(Validator):
         return isinstance(value, str)
 
     cdef bint is_array_typed(self) except -1:
-        return issubclass(self.dtype.type, np.str_)
+        return cnp.PyDataType_ISSTRING(self.dtype)
```

Comment:
So this is also wrong, `ISSTRING` checks for bytes (NPY_STRING) OR unicode.  You want `== NPY_UNICODE` here if you use this.